### ISSUE TITLE
Add --skip-frames (action repeat) support to train.py CLI

### ISSUE THREAD

**Issue by AlexandreEDMOND** [AlexandreEDMOND/numpy-rl-racer#126]
## Context
The `ActionRepeatEnv` wrapper is fully implemented in `src/numpy_rl_racer/env/wrappers.py` with 10+ passing tests covering reward accumulation, early termination, and all track types. This implements the "Frame skipping & action repeat" roadmap item from the README. However, the training script (`scripts/train.py`) does not expose it via CLI — users would need to modify the script manually to use it.

## Implementation scope
- Add `--skip-frames` (int, default=1) argument to `scripts/train.py`.
- When `--skip-frames > 1`, wrap the `RacingEnv` with `ActionRepeatEnv(env, skip_frames=N)`.
- The `ActionRepeatEnv` wrapper naturally handles `__getattr__` passthrough for `.track`, `.obstacles`, etc., so the rest of the training loop should work unchanged.
- The wrapper does not affect `state_dim` or observation shape, so no changes needed there.
- Add to `test_scripts.py`:
  - `test_train_skip_frames_env_wrapped` — verify the env is wrapped in `ActionRepeatEnv` when `--skip-frames 4` is passed.
  - `test_train_skip_frames_default_1` — verify default does not wrap.
  - `test_train_skip_frames_invalid` — verify `--skip-frames 0` is rejected.
  - `test_train_skip_frames_with_obstacles` — verify compatibility with `--num-obstacles`.

## Acceptance criteria
- [ ] `uv run python scripts/train.py --skip-frames 4 --episodes 1 --max-steps 5` succeeds.
- [ ] All existing tests pass: `uv run pytest`.
- [ ] New tests pass: `uv run pytest tests/test_scripts.py -k skip_frames`.
- [ ] `uv run ruff check .` reports no new issues.

## Suggested tests
```python
def test_train_skip_frames_env_wrapped(tmp_path):
    main = _make_main()
    wraps = []
    real_init = RacingEnv.__init__
    def tracking_init(self, **kwargs):
        wraps.append(kwargs)
        real_init(self, **kwargs)
    with patch.object(RacingEnv, "__init__", tracking_init), \
         patch.object(DQNAgent, "act", return_value=0), \
         patch.object(DQNAgent, "train_step", return_value=0.0), \
         patch.object(DQNAgent, "save"):
        main([
            "--skip-frames", "4",
            "--episodes", "1", "--max-steps", "1",
            "--save-dir", str(tmp_path),
        ])
    # Env was created (wraps has 1 entry for the inner env), then wrapped
    assert len(wraps) >= 1

def test_train_skip_frames_default_1(tmp_path):
    main = _make_main()
    with patch.object(DQNAgent, "act", return_value=0), \
         patch.object(DQNAgent, "train_step", return_value=0.0), \
         patch.object(DQNAgent, "save"):
        main([
            "--episodes", "1", "--max-steps", "1",
            "--save-dir", str(tmp_path),
        ])
    # Default skip_frames=1 means no wrapping — env is a RacingEnv, not ActionRepeatEnv

def test_train_skip_frames_with_obstacles(tmp_path):
    main = _make_main()
    with patch.object(DQNAgent, "act", return_value=0), \
         patch.object(DQNAgent, "train_step", return_value=0.0), \
         patch.object(DQNAgent, "save"):
        main([
            "--skip-frames", "3",
            "--num-obstacles", "2", "--obstacle-seed", "7",
            "--episodes", "1", "--max-steps", "1",
            "--save-dir", str(tmp_path),
        ])
```
